### PR TITLE
Add `MATCH` function

### PR DESCRIPTION
**Pull request by HactarCE**
Depends on `next-connections` branch

### CODE REVIEW

File: quadratic-core/src/formulas/functions/lookup.rs [hunk 1]
```diff
@@ -1026,4 +1081,84 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_match() {
```
Comment (ddimaria) [inline quadratic-core/src/formulas/functions/lookup.rs:1086]: Great test 🚀

File: quadratic-core/src/formulas/functions/lookup.rs [hunk 2]
```diff
@@ -274,10 +273,66 @@ fn get_functions() -> Vec<FormulaFunction> {
                 Array::new_row_major(result_size, final_output_array)?
             }
         ),
+        formula_fn!(
+            /// Searches for a value in a range and returns the index of the
```
Comment (ddimaria) [inline quadratic-core/src/formulas/functions/lookup.rs:277]: @luke-quadratic tagging you to review this documentation.

**Review by ddimaria — APPROVED** [review]
Great PR!

File: quadratic-core/src/formulas/functions/lookup.rs (hunk 2 above)
Comment (HactarCE) [reply quadratic-core/src/formulas/functions/lookup.rs:277]: Use `cargo run --bin docgen` to generate markdown

**Pull request merged**